--- /usr/share/emacs/site-lisp/elpa-src/markdown-mode-2.7/markdown-mode.el	2025-04-06 09:19:39.000000000 +0200
+++ /usr/share/emacs/site-lisp/elpa-src/markdown-mode-2.7/markdown-mode.el	2025-10-30 19:27:44.553151907 +0100
@@ -6832,8 +6832,8 @@
             (forward-line -1))
           (forward-line))
          ;; List items
-         ((setq bounds (markdown-cur-list-item-bounds))
-          (goto-char (nth 0 bounds)))
+         ;((setq bounds (markdown-cur-list-item-bounds))
+         ; (goto-char (nth 0 bounds)))
          ;; Other
          (t
           (while (and (not (bobp))
