--- /usr/share/emacs/site-lisp/elpa-src/markdown-mode-2.5/markdown-mode.el	2022-05-17 13:51:02.000000000 +0200
+++ /usr/share/emacs/site-lisp/elpa-src/markdown-mode-2.5/markdown-mode.el	2023-12-09 08:26:36.017296965 +0100
@@ -6487,8 +6487,8 @@
             (forward-line -1))
           (forward-line))
          ;; List items
-         ((setq bounds (markdown-cur-list-item-bounds))
-          (goto-char (nth 0 bounds)))
+         ;((setq bounds (markdown-cur-list-item-bounds))
+         ; (goto-char (nth 0 bounds)))
          ;; Other
          (t
           (while (and (not (bobp))
